easygcm-sample/src/main/java/eu/inloop/easygcm/sample/App.java

```diff
@@ -4,22 +4,20 @@
 import android.os.Bundle;
 
 import eu.inloop.easygcm.GcmListener;
-import eu.inloop.easygcm.WakeLockRelease;
 
 public class App extends Application implements GcmListener {
 
     @Override
-    public void onMessage(String s, Bundle bundle, WakeLockRelease wakeLockRelease) {
-        System.out.println("### message: " + s);
+    public void onMessage(String from, Bundle data) {
```

Comment:
+1. But the approach will be a bit different on GCM 3.0. The GcmReceiver is part of the GCM library and this is starting a wakeful service - which is implemented as EasyGcmListenerService. And it seems the implementation from Google is calling GcmReceiver.completeWakefulIntent() right inside the onStartCommand super implementation - which is kind of strange - I don't see a direct way how we can release the lock ourselves... ?